controllers/storageconsumer/storageconsumer_upgrade_controller.go

```diff
@@ -219,6 +220,15 @@ func (r *StorageConsumerUpgradeReconciler) reconcileStorageRequest(
 	rbdStorageRequest.SetGroupVersionKind(ocsv1alpha1.GroupVersion.WithKind("StorageRequest"))
 	rbdStorageRequest.Name = rbdStorageRequestName
 	rbdStorageRequest.Namespace = storageConsumer.Namespace
+	if err := r.Client.Get(ctx, client.ObjectKeyFromObject(rbdStorageRequest), rbdStorageRequest); client.IgnoreNotFound(err) != nil {
+		return err
+	}
+	rbdStorageRequestCopy := &metav1.PartialObjectMetadata{}
+	rbdStorageRequest.DeepCopyInto(rbdStorageRequestCopy)
+	rbdStorageRequestCopy.Finalizers = []string{}
+	if err := r.Client.Patch(ctx, rbdStorageRequestCopy, client.MergeFrom(rbdStorageRequest)); client.IgnoreNotFound(err) != nil {
+		return err
+	}
 	if err := r.Client.Delete(ctx, rbdStorageRequest); client.IgnoreNotFound(err) != nil {
 		return err
 	}
```

Comment:
pls move this into `else if rbdStorageRequest.UID != ""`, we don't want to ignore not found in 3 places.